ie_serving/models/ir_engine.py

```diff
@@ -140,9 +123,98 @@ def set_keys(self, mapping_config):
+        reshape_required = False
+        inputs_shapes = {}
+        for input_name, input_data in data.items():
+            net_input_shape = tuple(self.net.inputs[input_name].shape)
+            inputs_shapes[input_name] = input_data.shape
+            if net_input_shape != input_data.shape:
+                logger.debug("[Model: {}, version: {}] --- Shape change "
+                             "required for input: {}. Current "
+                             "shape: {}. Required shape: {}"
+                             .format(self.model_name, self.model_version,
+                                     input_name, net_input_shape,
+                                     input_data.shape))
+                reshape_required = True
+        return reshape_required, inputs_shapes
+
+    def reshape(self, reshape_param):
+        if type(reshape_param) is dict:
+            return self._reshape(reshape_param)
+        elif type(reshape_param) is int:
+            return self._change_batch_size(reshape_param)
+        else:
+            message = "Unknown error occurred in input reshape preparation"
+            logger.debug("[Model: {}, version: {}] --- {}".format(
+                self.model_name, self.model_version, message))
+            return True, message
+
+    def _reshape(self, inputs_shapes: dict):
+        #   Takes dictionary of input_name:shape pairs as parameter
+        #   (obtained from scan_input_shapes method)
+        #   Returns error status and error message. If no error occurred
+        #   returns False, None
+        logger.debug("[Model: {}, version: {}] --- Reshaping "
+                     "network...".format(self.model_name, self.model_version))
+        try:
+            self.net.reshape(inputs_shapes)
+        except Exception as e:
+            message = "Error occurred while reshaping: {}".format(str(e))
+            logger.debug("[Model: {}, version: {}] --- {}".format(
+                self.model_name, self.model_version, message))
+            return True, message
+        logger.debug("[Model: {}, version: {}] --- Reshaped successfully".
+                     format(self.model_name, self.model_version))
+
+        logger.debug("[Model: {}, version: {}] --- Loading network...".
+                     format(self.model_name, self.model_version))
+        try:
             self.exec_net = self.plugin.load(network=self.net)
-        results = self.exec_net.infer(inputs=data)
-        return results
+        except Exception as e:
+            message = "Error occurred while loading network: {}".format(
+                str(e))
+            logger.debug("[Model: {}, version: {}] --- {}".format(
+                self.model_name, self.model_version, message))
+            return True, message
+        logger.debug("[Model: {}, version: {}] --- Network loaded "
+                     "successfully".format(self.model_name,
+                                           self.model_version))
+        return False, None

```

Comment:
Like in `predict_utils.py` file, return only `message` or `None` should be enough